### PR TITLE
[GlobalISel] Fold G_SHUFFLE_VECTOR with a single element mask to G_EXTRACT_VECTOR_ELT

### PR DESCRIPTION
**Pull request by dzhidzhoev**
It introduces minor regression in arm64-vcvt_f.ll, which will be fixed later.

### CODE REVIEW

File: llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp
Comment (jayfoad) [inline llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:407]: You don't really need to check things that are already enforced by MachineVerifier. All you need here is `Mask.size() == 1 && Mask[0] >= 0`.

File: llvm/test/CodeGen/AArch64/ext-narrow-index.ll
Comment (davemgreen) [inline llvm/test/CodeGen/AArch64/ext-narrow-index.ll:3]: This can use  `--check-prefixes=CHECK,CHECK-SD` for the first line and ` --check-prefixes=CHECK,CHECK-GI` for the second. Where they are identical the check lines should then fold to both using `CHECK`

File: llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp
Comment (jayfoad) [inline llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:409]: You should be able to build instructions that write to DstReg, instead of creating NewDstReg and then calling replaceRegWith. Unless I'm missing something.

File: llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp
Comment (jayfoad) [inline llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:403]: Maybe also handle the Mask[0] == -1 case, by replacing it with G_IMPLICIT_DEF ?

File: llvm/test/CodeGen/AArch64/GlobalISel/prelegalizercombiner-shuffle-vector.mir
Comment (jayfoad) [inline llvm/test/CodeGen/AArch64/GlobalISel/prelegalizercombiner-shuffle-vector.mir:15]: Could you please pre-commit a patch which just updates the checks in this file (no review needed) and then rebase the current patch on top of that?

**Review by jayfoad — APPROVED** [review]
LGTM if the AArch64 maintainers are OK with the slight regression.

**Comment by davemgreen** [comment]
Yeah I agree, this LGTM. The fcvt2 are being fixed in a followup.

**Pull request closed without merging**
